### PR TITLE
Add support for serialization via `serde`

### PR DESCRIPTION
**Pull request by pcwalton**
r? @SimonSapin

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -25,6 +25,7 @@ rand = "0"
 phf = "0.7"
 phf_macros = "0.7"
 lazy_static = "0.1.10"
+serde = "*"
```
Comment (SimonSapin) [inline Cargo.toml]: Version number please. Cargo takes it as "anything SemVer-compatible with this".

File: src/atom/mod.rs [hunk 2]
```diff
@@ -309,6 +310,20 @@ impl AsRef<str> for Atom {
...
+impl Serialize for Atom {
+    fn serialize<S>(&self, serializer: &mut S) -> Result<(),S::Error> where S: Serializer {
+        let string: &str = self.as_ref();
+        string.serialize(serializer)
+    }
+}
+
+impl Deserialize for Atom {
+    fn deserialize<D>(deserializer: &mut D) -> Result<Atom,D::Error> where D: Deserializer {
+        let string: String = try!(Deserialize::deserialize(deserializer));
```
Comment (SimonSapin) [inline src/atom/mod.rs]: It's `String` here and `str` above. Is this OK?

Comment (pcwalton) [inline src/atom/mod.rs]: Should be, yeah.

**Comment by SimonSapin** [comment]
r=me with `serde = "0.4.2"` in `Cargo.toml`.

**Pull request merged**